File: csharp/src/ev1/ut2_controlflujometodos/u05repaso/ejercicios/resueltos/E03_RepasoSwitchMenu_Resuelto.cs
/*
OBJETIVO: Switch con opcion 1-3: Nuevo, Abrir, Salir.
SOLUCION: ver codigo.
PATRON: Menu do-while con opcion 0 salir y uso de Console.ReadLine().
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/
using System;
class Program
{
    static void Main()
    {
        string? opcion;
        do
        {
            Console.WriteLine();
            Console.WriteLine("=== E03 Repaso Switch Menu Resuelto ===");
            Console.WriteLine("1. Ejecutar ejemplo con opcion 2");
            Console.WriteLine("2. Elegir una opcion del switch");
            Console.WriteLine("0. Salir");
            Console.Write("Elige una opcion: ");
            opcion = Console.ReadLine();
            switch (opcion)
            {
                case "1":
                    EjecutarSwitch(2);
                    break;
                case "2":
                    Console.Write("Opcion interna (1-3): ");
                    int opInterna = int.Parse(Console.ReadLine() ?? "0");
                    EjecutarSwitch(opInterna);
                    break;
                case "0":
                    Console.WriteLine("Saliendo...");
                    break;
                default:
                    Console.WriteLine("Opcion no valida.");
                    break;
            }
        } while (opcion != "0");
    }
    static void EjecutarSwitch(int op)
    {
        switch (op)
        {
            case 1:
                Console.WriteLine("Nuevo");
                break;
            case 2:
                Console.WriteLine("Abrir");
                break;
            case 3:
                Console.WriteLine("Salir");
                break;
            default:
                Console.WriteLine("Invalida");
                break;
        }
    }
}

File: cpp/src/ev1/ut2_controlflujometodos/u05repaso/ejercicios/resueltos/E03_RepasoSwitchMenu_Resuelto.cpp
/*
OBJETIVO: Switch con opcion 1-3: Nuevo, Abrir, Salir.
SOLUCION: ver codigo.
PATRON: Menu do-while con opcion 0 salir y uso de cin.
Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
using namespace std;

void ImprimirMenu() {
    cout << endl;
    cout << "=== E03 Repaso Switch Menu Resuelto ===" << endl;
    cout << "1. Ejecutar ejemplo con opcion 2" << endl;
    cout << "2. Elegir una opcion del switch" << endl;
    cout << "0. Salir" << endl;
}

void EjecutarSwitch(int op)
{
    switch (op)
    {
        case 1:
            cout << "Nuevo" << endl;
            break;
        case 2:
            cout << "Abrir" << endl;
            break;
        case 3:
            cout << "Salir" << endl;
            break;
        default:
            cout << "Invalida" << endl;
            break;
    }
}

int main() {
    int opcion;
    do {
        ImprimirMenu();
        cout << "Introduce una opcion -> ";
        cin >> opcion;
        switch (opcion) {
            case 1:
                EjecutarSwitch(2);
                break;
            case 2: {
                int opInterna;
                cout << "Opcion interna (1-3): ";
                cin >> opInterna;
                EjecutarSwitch(opInterna);
                break;
            }
            case 0:
                cout << "Saliendo..." << endl;
                break;
            default:
                cout << "Opcion no valida." << endl;
                break;
        }
    } while (opcion != 0);
    return 0;
}
